### PR TITLE
[#644] Instance Properties Step: Add warning for flavors which may not be big enough

### LINKED ISSUE #644: [INSTANCE PROPERTIES STEP] Warn if chosen flavor may not be big enough
***Follow up task for Instance Properties Step***

Warn if the user edits a row in the Instance Properties Step Table to choose a flavor that is not "big enough", i.e., the chosen flavor is neither the `best fit flavor` nor does it exist in the `all fit` collection

### PR DESCRIPTION
**Pull request by mturley**
Closes #644.

screenshot 2018-09-26 13 29 08

screenshot 2018-09-26 13 29 21

screenshot 2018-09-26 13 29 38

### CODE REVIEW

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/helpers.js [hunk 1]
```diff
@@ -31,3 +31,14 @@ export const getVmsWithTargetClusterName = (vms, destinationTenantIdsBySourceClu
       target_cluster_name: tenant ? tenant.name : ''
     };
   });
+
+export const allFitForVM = (bestFitFlavors, tenantFlavors, vmId) => {
+  const flavorsSlugPrefix = 'flavors/';
+  const flavorForVM = bestFitFlavors.find(flavor => flavor.source_href_slug === `vms/${vmId}`);
+  const allFitFlavors = flavorForVM && flavorForVM.all_fit;
+  if (!allFitFlavors || allFitFlavors.length === 0) {
+    return [tenantFlavors.reduce((prev, current) => (prev.root_disk_size > current.root_disk_size ? prev : current))];
```
Comment (michaelkro) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/helpers.js]: Do we want to switch to memory/cpu here as well?

Comment (michaelkro) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/helpers.js]: Reading through this one more time, this reads as if

> if there are no all fit flavors, then find the "biggest flavor" and do not mark it with an asterisk

Is this desired?  I think all flavors that are too small should be marked as a warning

Comment (mturley) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/helpers.js]: Ah, that's true. I borrowed this helper code from Aparna's PR here: https://github.com/ManageIQ/manageiq-v2v/pull/655/files, really it should just return nothing if there is no all fit flavor. I'll take a closer look.

Comment (mturley) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/helpers.js]: This should be fixed in 7644f6b.

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/PlanWizardInstancePropertiesStepReducer.js [hunk 2]
```diff
@@ -72,7 +72,11 @@ export default (state = initialState, action) => {
         } else {
           bestFitFlavor =
             tenantFlavors &&
-            tenantFlavors.reduce((prev, current) => (prev.root_disk_size > current.root_disk_size ? prev : current));
+            tenantFlavors.reduce((prev, current) => {
+              if (prev.memory > current.memory) return prev;
+              if (prev.memory < current.memory) return current;
+              return prev.cpus > current.cpus ? prev : current;
```
Comment (AparnaKarve) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/PlanWizardInstancePropertiesStepReducer.js]: We can revisit (and remove) this if/when the backend gives us this info.

But for now, this looks good!

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/components/PlanWizardInstancePropertiesStepTable.js [hunk 3]
```diff
@@ -170,6 +202,29 @@ class PlanWizardInstancePropertiesStepTable extends React.Component {
...
     });
 
+    const flavorColumnPopoverContent = (
+      <Popover id="v2v-osp-flavor-info" className="popover-with-max-width">
+        <p>
+          {__('In OpenStack, flavors define the compute, memory, and storage capacity of nova computing instances.')}
+        </p>
+        <p>
+          {__(
+            'A flavor name appended with an * indicates that the flavor has CPU or memory capacity smaller than the source VM. Selecting these flavors might cause problems when the VM is migrated.'
```
Comment (AparnaKarve) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/components/PlanWizardInstancePropertiesStepTable.js]: Can you get the entire gettext expression in a single line and then add the `// prettier-ignore` comment?

**Comment by AparnaKarve** [comment]
Tested and verified that OSP Plans are successfully created.

Nice and clean implementation!
Thanks @mturley !!

**Comment by AparnaKarve** [comment]
Other than the gettext adjustment that is needed above, everything else looks great!

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 7def8964ae6336e87a0771262e77539beefadd44
Author: Aparna Karve <aparna.karve@gmail.com>
Date:   Thu Sep 27 14:42:29 2018 -0700

    Merge pull request #669 from mturley/644-warn-chosen-flavor
    
    [#644] Instance Properties Step: Add warning for flavors which may not be big enough
    
    (cherry picked from commit c25064e3e847079dabfbe75e5414f33dc3c9b485)
```

**Pull request merged**